crates/ruff_formatter/src/printer/mod.rs

```diff
@@ -311,6 +314,14 @@ impl<'a> Printer<'a> {
                     _ => {
                         self.state.measured_group_fits = true;
 
+                        // Set the print mode for this group in case this group was measured as part of a
+                        // previous `fits_group` call.
+                        if let Some(id) = id {
```

Comment:
This was a bug in the Printer where group modes set during a previous `fits` test stuck around.